envoy/upstream/host_description.h

```diff
@@ -55,6 +55,48 @@ struct HostStats {
   }
 };
 
+/**
+ * Weakly-named load metrics to be reported as EndpointLoadMetricStats. Individual stats are
+ * accumulated by calling add(), which combines stats with the same name. The aggregated stats are
+ * retrieved by calling latch(), which also clears the current load metrics.
+ */
+class LoadMetricStats {
+public:
+  struct Stat {
+    uint64_t num_requests_with_metric = 0;
+    double total_metric_value = 0.0;
+  };
+
+  using StatsMap = absl::flat_hash_map<std::string, Stat>;
+
+  LoadMetricStats() : map_(std::make_unique<StatsMap>()) {}
+
+  // Adds the given stat to the map. If the stat already exists in the map, then the stat is
+  // combined with the existing map entry by incrementing num_requests_with_metric and summing the
+  // total_metric_value fields.
+  void add(const std::string& key, double value) {
+    absl::MutexLock lock(&mu_);
+    Stat& stat = (*map_)[key];
```

Comment:
One thing to note is that the name is shared between different extensions (multiple extensions can add to the same key). Is this intended?